index.js

```diff
@@ -2613,6 +2613,42 @@
   }
   S.ifElse = def('ifElse', {}, [Pred(a), Fn(a, b), Fn(a, b), a, b], ifElse);
 
+  //# when :: (a -> Boolean) -> (a -> b) -> a -> b
```

Comment:
> What is wrong with `when :: (a -> a) -> a -> a`, @cust0dian?

@davidchambers sorry, not sure I follow that signature: `when :: (a -> a) -> a -> a` looks like `id` to me 😄 

Did you mean `when :: (a -> Bool) -> (a -> a) -> a -> a`? In that case there's nothing wrong with it, I was just commenting that it's the only correct type if we follow definitions of `ifElse` and `id`